### PR TITLE
Dynamically detect MiniYAML files

### LINKED ISSUE #2: Red squiggly lines everywhere
image

### PR DESCRIPTION
**Pull request by letmaik**
Fixes #2.

Inspired by https://github.com/microsoft/vscode/issues/135965#issuecomment-957164770.

### CODE REVIEW

File: clients/oraide-vscode/src/extension.ts [hunk 1]
```diff
@@ -72,11 +82,11 @@ function start(context: vscode.ExtensionContext, serverPath: string, workspaceFo
         // Register the server for 'yaml' (.yaml) files. This uses the definition for 'miniyaml' found in package.json under 'contributes.languages'. (not any more)
```
Comment (abcdefg30) [inline clients/oraide-vscode/src/extension.ts]:
```suggestion
        // Register the server for 'yaml' (.yaml) files. This uses the definition for 'miniyaml' found in package.json under 'contributes.languages'.
```
?

File: clients/oraide-vscode/src/extension.ts [hunk 2]
```diff
@@ -46,6 +46,16 @@ export async function activate(context: vscode.ExtensionContext) {
     }
 
     start(context, serverPath, workspaceFolderPath, gameFolderPath!);
+
+    // Detect if a file identified as YAML is actually MiniYAML and switch if so.
+    vscode.workspace.onDidOpenTextDocument(e => {
+        if (e.languageId == 'yaml') {
+            const text = e.getText();
+            if (text.includes('\t')) {
```
Comment (abcdefg30) [inline clients/oraide-vscode/src/extension.ts:54]: I wonder if this is even necessary, considering there shouldn't be any "real" yaml files inside an OpenRA workspace.(?)

Comment (letmaik) [reply clients/oraide-vscode/src/extension.ts:54]: Well, there are a few, e.g. GitHub workflow files and tool configs. It's a conservative approach that tries to be minimally disruptive. If someone uses spaces for indentation of MiniYAML then they can still manually switch the language per file, or alternatively assign all '.yaml' files through the workspace `settings.json`:

```json
    "files.associations": {
        "*.yaml": "miniyaml"
    }
```

Comment (Mailaender) [reply clients/oraide-vscode/src/extension.ts:54]: CI is configured using YAML yaml.

Comment (abcdefg30) [reply clients/oraide-vscode/src/extension.ts:54]: Ah good point.

**Comment by penev92** [comment]
Thanks!
This works as advertised, but I have one small nit - e0213d41c476a0aec3f87c72efce45039e8b8e22 disabled the custom syntax file for a reason. With it enabled, section folding changes to include the empty line between nodes, resulting in this:
image
(and who knows what else)
Would you by any chance have an idea how to fix that?

**Comment by letmaik** [comment]
> With it enabled, section folding changes to include the empty line between nodes
    Would you by any chance have an idea how to fix that?

Yeah, this was the missing language configuration file which I added. The docs at https://code.visualstudio.com/api/language-extensions/language-configuration-guide don't document `"offSide"` yet, but that's what the built-in YAML language extension uses as well: https://github.com/microsoft/vscode/blob/a2348fa5f200818e3ebe118fc431dc642cee18d3/extensions/yaml/language-configuration.json#L25

**Comment by penev92** [comment]
Ah, thank you!

**Comment by penev92** [comment]
@letmaik something I forgot to mention - the `onDidOpenTextDocument` event isn't triggered for the file that is open and focused when you reopen/reload a workspace. Do you maybe have a workaround for that?

**Pull request merged**